src/Models/ShallowWaterModels/shallow_water_diffusion_operators.jl

```diff
@@ -0,0 +1,81 @@
+using Oceananigans.Operators
+using Oceananigans.Architectures: device, device_event
+using Oceananigans.TurbulenceClosures: ExplicitTimeDiscretization, ThreeDimensionalFormulation
+
+using Oceananigans.TurbulenceClosures: 
+                        AbstractScalarDiffusivity, 
+                        convert_diffusivity,
+                        viscosity_location, 
+                        viscosity, 
+                        ν_σᶜᶜᶜ,
+                        ∂ⱼ_τ₁ⱼ,
+                        ∂ⱼ_τ₂ⱼ
+
+import Oceananigans.TurbulenceClosures:
+                        DiffusivityFields,
+                        calculate_diffusivities!,
+                        calculate_nonlinear_viscosity!, 
+                        viscosity,
+                        with_tracers,
+                        νᶜᶜᶜ
+
+struct ShallowWaterScalarDiffusivity{N, X} <: AbstractScalarDiffusivity{ExplicitTimeDiscretization, ThreeDimensionalFormulation}
+    ν :: N
+    ξ :: X
+end
+
+function ShallowWaterScalarDiffusivity(FT::DataType=Float64; ν=0, ξ=0, discrete_form = false) 
+    ν = convert_diffusivity(FT, ν; discrete_form)
+    ξ = convert_diffusivity(FT, ξ; discrete_form)
+    return ShallowWaterScalarDiffusivity(ν, ξ)
+end
+
+# We have no tracers in the shallow water diffusivity
+with_tracers(tracers, closure::ShallowWaterScalarDiffusivity) = closure
+viscosity(closure::ShallowWaterScalarDiffusivity, K) = closure.ν
+
+# The diffusivity for the shallow water model is calculated as h*ν in order to have a viscous term in the form
+# h⁻¹ ∇ ⋅ (hν t) where t is the 2D stress tensor plus a trace => t = ∇u + (∇u)ᵀ - ξI⋅(∇⋅u)
+
+@inline νᶜᶜᶜ(i, j, k, grid, closure::ShallowWaterScalarDiffusivity, clock, solution, C) = 
+        solution.h[i, j, k] * νᶜᶜᶜ(i, j, k, grid, clock, viscosity_location(closure), closure.ν)
+
+function calculate_diffusivities!(diffusivity_fields, closure::ShallowWaterScalarDiffusivity, model)
+    arch = model.architecture
+    grid = model.grid
+    solution = model.solution
+    clock = model.clock
+    
+    event = launch!(arch, grid, :xyz,
+                    calculate_nonlinear_viscosity!,
+                    diffusivity_fields.νₑ, grid, closure, clock, solution, nothing,
+                    dependencies = device_event(arch))
+
+    wait(device(arch), event)
+
+    return nothing
+end
+
+DiffusivityFields(grid, tracer_names, bcs, ::ShallowWaterScalarDiffusivity)  = (; νₑ=CenterField(grid, boundary_conditions=bcs.h))
+
+#####
+##### Diffusion flux divergence operators
+#####
+
+@inline shallow_∂ⱼ_τ₁ⱼ(i, j, k, grid, closure, K, clock, velocities, tracers, solution, ::ConservativeFormulation) = 
```

Comment:
If we have arguments of this function to be either ConservativeFormulation or VectorInvariantFormulation, do we need the `shallow_` in front or can we drop it and know that dispatch will use the right one?  Or, if we need something, maybe sw for shallow water?  Just a thought.

Reply:
just exporting `∂ⱼ_τ₁ⱼ` for the formulation? That can work

Reply:
Maybe it's better to add on something, like you did, as it will make it clearer to the user that it's a different function.  `shallow` is not something that we have used before.  `ShallowWaterModel` is too long.  I wonder if `SWM`, might be a good choice?  We can also keep this as is and polish it off later, after we know things are working well.